decidim-participatory_processes/lib/decidim/participatory_processes/engine.rb

```diff
@@ -12,6 +12,16 @@ class Engine < ::Rails::Engine
       isolate_namespace Decidim::ParticipatoryProcesses
 
       routes do
+        get "processes/:process_id", to: redirect { |params, _request|
+          process = Decidim::ParticipatoryProcess.where(id: params[:process_id]).first
+          process ? "/processes/#{process.slug}" : "/404"
+        }, constraints: { process_id: /[0-9]+/ }
+
+        get "/processes/:process_id/f/:feature_id", to: redirect { |params, _request|
+          process = Decidim::ParticipatoryProcess.where(id: params[:process_id]).first
+          process ? "/processes/#{process.slug}/f/#{params[:feature_id]}" : "/404"
+        }, constraints: { process_id: /[0-9]+/ }
```

Comment:
@mrcasals Maybe you can use `find(params[:process_id])` here. If the record is not found, an `ActiveRecord::RecordNotFound` error will be raised, which will result in a 404. I think that's better than manually redirecting to the "/404" page, since in that case the request will have a 200 status instead of a 404.

Also, to be technically correct, I guess this should be a fallback and the preferred routes should be matched first? I'm think of the super-unlikely case of numerical slugs.

Reply:
Rails routes run the first definition that matches, so I need these redirections on top since they have the constraint.

Reply:
Right, I see. The other alternative would be to manually redirect from the controller instead. But numerical slugs is a small edge case anyways.